### PR TITLE
Fix A4A issues #5472 & #5273

### LINKED ISSUE #5273: Move `incrementLoadingAds` to a later time for a4a ads
As discussed, there's no need to block 3P ads rendering when the a4a ad response is on its way, instead, we can just do it right before creating the 3P ad iframe.

This will also simplify code by removing the logic that forces `decrementLoadingAds`.

### LINKED ISSUE #5472: A4A resolve layoutCallback early when renderViaCrossDomainIframe
I found that if an A4A decide to render as 3p ad it will call `this.renderViaCrossDomainIframe_` and resolve the `#layoutCallback` immediately after that. 

However in 3PAdImpl, we will only resolve `#layoutCallback` when this `apiHandler.startUp` promise is resolved. https://github.com/ampproject/amphtml/blob/master/extensions/amp-ad/0.1/amp-ad-3p-impl.js#L217

I think we should keep the timestamp same in those two cases.

### PR DESCRIPTION
**Pull request by keithwrightbos**
Issues addressed:
- Fix #5472: ensure AmpAdXOriginIframeHandler.init promise is returned as part of layoutCallback
- Fix #5273: only increment 3p ad throttling when creative known not to be AMP

### CODE REVIEW

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 1]
```diff
@@ -240,6 +236,13 @@ export class AmpA4A extends AMP.BaseElement {
    * @override
    */
   preconnectCallback(unusedOnLayout) {
+    // TODO(tdrl): Temporary, while we're verifying whether SafeFrame is an
+    // acceptable solution to the 'Safari on iOS doesn't fetch iframe src from
+    // cache' issue.  See https://github.com/ampproject/amphtml/issues/5614
+    this.preconnect.url(SAFEFRAME_IMPL_PATH);
+    //if (!this.config) {
```
Comment (tdrl) [inline extensions/amp-a4a/0.1/amp-a4a.js]: If you're commenting this out, I'm not sure you need the whole rearrangement of this function.

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 2]
```diff
@@ -448,6 +447,7 @@ export class AmpA4A extends AMP.BaseElement {
    * @private
    */
   promiseErrorHandler_(error) {
+    console.log('promiseErrorHandler_', error);
```
Comment (tdrl) [inline extensions/amp-a4a/0.1/amp-a4a.js]: You'll have to get rid of this for real submission, or convert it to a `user().info()` or something.

File: extensions/amp-a4a/0.1/test/test-amp-a4a.js [hunk 3]
```diff
@@ -30,12 +30,16 @@ import {data as testFragments} from './testdata/test_fragments';
 import {data as expectations} from './testdata/expectations';
 import {installDocService} from '../../../../src/service/ampdoc-impl';
 import {a4aRegistry} from '../../../../ads/_a4a-config';
-import '../../../../extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler';
+import {AmpAdXOriginIframeHandler}
+  from '../../../../extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler';
 import * as sinon from 'sinon';
 
+const XHR_URL =  'http://iframe.localhost:' + location.port +
```
Comment (tdrl) [inline extensions/amp-a4a/0.1/test/test-amp-a4a.js]: Do you need the `iframe` part of that URL?

File: extensions/amp-a4a/0.1/amp-a4a.js (hunk 1 above)
Comment (keithwrightbos) [inline extensions/amp-a4a/0.1/amp-a4a.js]: Didn't mean to comment out...

File: extensions/amp-a4a/0.1/amp-a4a.js (hunk 2 above)
Comment (keithwrightbos) [inline extensions/amp-a4a/0.1/amp-a4a.js]: Removed all console.logs

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 4]
```diff
@@ -240,6 +236,13 @@ export class AmpA4A extends AMP.BaseElement {
    * @override
    */
   preconnectCallback(unusedOnLayout) {
+    // TODO(tdrl): Temporary, while we're verifying whether SafeFrame is an
+    // acceptable solution to the 'Safari on iOS doesn't fetch iframe src from
+    // cache' issue.  See https://github.com/ampproject/amphtml/issues/5614
+    this.preconnect.url(SAFEFRAME_IMPL_PATH);
+    if (!this.config) {
```
Comment (jridgewell) [inline extensions/amp-a4a/0.1/amp-a4a.js]: Isn't `#config` guaranteed?

**Comment by lannka** [comment]
@keithwrightbos we agreed on bumping AMP creative priority to 1 so that it will be rendered before 3p ads. But I don't remember if we want to make AMP creative exempted from the "one ad a time" rule.

+ @cramforce  @jasti

**Comment by keithwrightbos** [comment]
My understanding was that was the intent from the start.

**Comment by keithwrightbos** [comment]
lannka@ latest commit fixed remaining lint and type check issues.  Can you please merge?  Thanks

**Comment by keithwrightbos** [comment]
@lannka added you incorrectly in last comment :D

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 5]
```diff
@@ -587,7 +586,9 @@ export class AmpA4A extends AMP.BaseElement {
    * testing.
    * @visibleForTesting
    */
-  onCrossDomainIframeCreated(iframe) {}
+  onCrossDomainIframeCreated(iframe) {
+    dev().info(`onCrossDomainIframeCreated ${iframe}`);
```
Comment (jridgewell) [inline extensions/amp-a4a/0.1/amp-a4a.js]: `TAG` must be the first parameter.

File: extensions/amp-a4a/0.1/amp-a4a.js (hunk 4 above)
Comment (keithwrightbos) [inline extensions/amp-a4a/0.1/amp-a4a.js]: This was requested by @tdrl as he had seen it not being present in the wild.  Adding sanity check can't hurt...

File: extensions/amp-a4a/0.1/amp-a4a.js (hunk 5 above)
Comment (keithwrightbos) [inline extensions/amp-a4a/0.1/amp-a4a.js]: Done

**Pull request merged**